### DISCUSSION TITLE
Cpp constructor initialization suggestion

### DISCUSSION THREAD

**Discussion by shemeshg** [hsutter/cppfront#1111] [Q&A]
```
MyClass: type = {
	private a: const int;
	operator=: (
        out this,
        a_in: int
		) == {
			a = a_in;
		}
}
```

I can see `cppfront` is claver and knows to `a{ a_in }{...` at the `cpp` file. I've been wondering maybe at the `cpp2` we can shorten that for constructors only to

```
MyClass: type = {
	private a: const int;
	operator=: (
			out this,
			a{a_in: int}
		) == {
		}
}
```

or

```
MyClass: type = {
	private a: const int;
	operator=: (
			out this,
			a{int}
		) == {
		}
}
```

or 
```
MyClass: type = {
	private a: const int;
	operator=: (
			out this,
			a
		) == {
		}
}
```

**Comment by hsutter** [comment]
Thanks!

> shorten that for constructors only

Cpp2 intentionally unifies construction and assignment under `operator=`, and intentionally removes special-purpose constructor grammar and semantics (like the mem-init-list). I think it's a feature that all parameters and all functions have the same grammar, so I don't want to create a special grammar for constructors only.

Thanks for understanding. If I misunderstood the suggestion, feel free to reopen!
